mono/metadata/threadpool.c

```diff
@@ -432,7 +432,11 @@ mono_threadpool_begin_invoke (MonoDomain *domain, MonoObject *target, MonoMethod
 
 	error_init (error);
 
-	message = mono_method_call_message_new (method, params, mono_get_delegate_invoke (method->klass), (params != NULL) ? (&async_callback) : NULL, (params != NULL) ? (&state) : NULL, error);
+	MonoMethod* invoke = NULL;
+	if (method->klass->parent == mono_defaults.multicastdelegate_class)
```

Comment:
Use `mono_class_is_delegate` ?

Reply:
That looks better. I did not know it existed. I'll use it instead.

Reply:
I've force pushed this change.